### PR TITLE
Avoid slash followed by invalid characters in wfopen for Windows

### PR DESCRIPTION
**Pull request by jotacarma90**
|Related issue|
|---|
|https://github.com/wazuh/internal-devel-requests/issues/744|

## Description

In this PR we will replace all fopen with wfopen, and add a small modification to avoid the misuse of paths starting with / or followed by invalid characters.
It is also necessary to modify all the unit tests that used these functions, and add some extra cases to check the new functionality.

## Configuration options

```
<agent_config>
    <localfile>
        <location>\\192.168.56.103\x\x</location>
        <log_format>syslog</log_format>
    </localfile>
</agent_config>
```

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
- [x] Source installation
- [x] Review logs syntax and correct language

### CODE REVIEW

File: src/shared/file_op.c [hunk 1]
```diff
@@ -2710,6 +2710,14 @@ FILE * wfopen(const char * pathname, const char * mode) {
     FILE * fp;
     int i;
 
+    if (pathname[0] == '\\' || pathname[0] == '/') {
```
Comment (lchico) [inline src/shared/file_op.c:2713]: Could you check if the pointer is valid before performing a comparison?

Comment (jotacarma90) [reply src/shared/file_op.c]: Done!

**Comment by jotacarma90** [comment]
## Testing

After configure a file in `logcollector`, I proceed with a `mv` command to change file inode, and I have checked the correct functionality of logcollector.

- Linux logcollector rotation file:

```
2024/02/14 10:57:18 wazuh-logcollector[194101] logcollector.c:504 at LogCollectorStart(): DEBUG: Performing file check.
2024/02/14 10:57:18 wazuh-logcollector[194101] logcollector.c:725 at LogCollectorStart(): DEBUG: File inode changed. /test/file.log
```
```
** Alert 1707908238.225054: - ossec,pci_dss_10.5.2,pci_dss_10.5.5,gpg13_10.1,gdpr_II_5.1.f,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.9,tsc_CC6.1,tsc_CC7.2,tsc_CC7.3,tsc_PI1.4,tsc_PI1.5,tsc_CC7.1,tsc_CC8.1,
2024 Feb 14 10:57:18 ubuntu23->logcollector
Rule: 591 (level 3) -> 'Log file rotated.'
ossec: File rotated (inode changed): '/test/file.log'.
```


- Windows logcollector rotation file:

```
2024/02/14 03:03:42 wazuh-agent[1116] logcollector.c:504 at LogCollectorStart(): DEBUG: Performing file check.
2024/02/14 03:03:42 wazuh-agent[1116] logcollector.c:725 at LogCollectorStart(): DEBUG: File inode changed. c:/test/file.tx
```

```
** Alert 1707908608.225407: - ossec,pci_dss_10.5.2,pci_dss_10.5.5,gpg13_10.1,gdpr_II_5.1.f,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.9,tsc_CC6.1,tsc_CC7.2,tsc_CC7.3,tsc_PI1.4,tsc_PI1.5,tsc_CC7.1,tsc_CC8.1,
2024 Feb 14 11:03:28 (windows11) any->logcollector
Rule: 591 (level 3) -> 'Log file rotated.'
ossec: File rotated (inode changed): 'c:/test/file.txt'.
```

**Review by vikman90 — APPROVED** [review]

**Review by mjcr99 — DISMISSED** [review]

**Comment by jotacarma90** [comment]
### PR checks failing
After research none of the failed checks seem to be related to the changes in this PR.
Here is a PR dummy of 4.8.0 that was done 5 days ago with similar errors:
https://github.com/wazuh/wazuh/pull/21857/checks

**Review by MarcelKemp — DISMISSED** [review]

**Review by MarcelKemp — APPROVED** [review]
GJ!

**Comment by xenomuta** [comment]
Amazing job guys!

**Pull request merged**